Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp

```diff
@@ -255,14 +255,28 @@ CacheStorageManager::CacheStorageManager(const String& path, CacheStorageRegistr
 
 CacheStorageManager::~CacheStorageManager()
 {
-    for (auto& request : m_pendingSpaceRequests)
+    close();
+}
+
+void CacheStorageManager::close()
+{
+    while (!m_pendingSpaceRequests.isEmpty()) {
+        auto request = m_pendingSpaceRequests.takeFirst();
         request.second(false);
+    }
 
     for (auto& cache : m_caches)
         m_registry.unregisterCache(cache->identifier());
+    m_caches.clear();
 
     for (auto& identifier : m_removedCaches.keys())
         m_registry.unregisterCache(identifier);
+    m_removedCaches.clear();
+    m_cacheRefConnections.clear();
+
+    makeDirty();
+    m_size = std::nullopt;
```

Comment:
What about m_pendingSize?

Reply:
Good point! I've reset m_pendingSize here. I also made it track the cache identifiers (so we only update its value for latest initialization).